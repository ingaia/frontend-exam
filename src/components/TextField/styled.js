import styled from 'styled-components'

export const TextFieldStyled = styled.div`
  	position: relative;
`;

export const Input = styled.input`
  font-size: 14px;
  width: 100%;
  padding: 1.5em 0 0.3em 0.2em;
  background: transparent;
  border: 0;
  border-bottom: 1px solid ${props => props.theme.palette.primary};
  color: ${props => props.theme.palette.white};
  
  :invalid {
  	box-shadow: none;
  }
  
`;

export const Label = styled.label`
  position: absolute;
  pointer-events: none;
  left: 5px;
  top: 25px;
  transition: 0.3s ease all;
  color: ${props => props.theme.palette.primary};
  text-transform: capitalize;
  
  ${Input}:valid ~ & {
    top: 6px;
    left: 5px;
    font-size: 11px;
  }
  
  ${Input}:focus ~ & {
    top: 6px;
    left: 5px;
    font-size: 11px;
  }
  
`;