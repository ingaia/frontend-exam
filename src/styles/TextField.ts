import { makeStyles } from '@material-ui/core/styles';

export const inputStyles = makeStyles(() => ({
  input: {
    color: 'var(--white)'
  },
  underline: {
    borderBottomColor: 'var(--hillary)',
    '&:before, &:after': {
      borderBottomColor: 'var(--hillary)',
      '&:hover': {
        borderBottomColor: 'red',
      }
    },
    '&:hover': {
      borderBottomColor: 'red',
      '&:before': {
        borderBottomColor: 'red',
      }
    }
  }
}));

export const labelStyles = makeStyles(() => ({
  root: {
    color: 'var(--hillary)'
  },
  focused: {
    color: 'var(--hillary) !important'
  }
}));
