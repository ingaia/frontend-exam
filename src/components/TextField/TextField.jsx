import React from 'react';
import PropTypes from 'prop-types';
import { TextFieldStyled, Input, Label } from './styled';

const TextField = ({id, onChange, value, type, label, className}) => {
	return(
		<TextFieldStyled className={className}>
			<Input id={id} onChange={onChange} value={value} type={type} required/>
			<Label htmlFor={id}>{label}</Label>
		</TextFieldStyled>
	)
};

export default TextField

TextField.propTypes = {
	onChange: PropTypes.func.isRequired,
  	type: PropTypes.string.isRequired,
  	value: PropTypes.string.isRequired,
		label: PropTypes.string
};