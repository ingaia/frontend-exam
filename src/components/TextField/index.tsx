import React from 'react';
import TextField from '@material-ui/core/TextField';

import { inputStyles, labelStyles } from '../../styles/TextField';

export interface Props {
  fullWidth?: boolean,
  id: string,
  type?: string,
  label: string,
  value?: string,
  onChange?: any;
}

const Component = (props: Props) => (
  <TextField
    {...props}
    InputProps={{
      classes: inputStyles()
    }}
    InputLabelProps={{
      classes: labelStyles()
    }}
  />
);

export default Component;
