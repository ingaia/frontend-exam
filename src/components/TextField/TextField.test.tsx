import React from 'react';
import { render } from '@testing-library/react';
import TextField from '.';

describe('<TextField />', () => {
  it('should render correctly', () => {
    const { container } = render(<TextField id="jest" label="jest" />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
