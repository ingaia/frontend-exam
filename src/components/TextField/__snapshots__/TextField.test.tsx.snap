// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> should render correctly 1`] = `
<div
  class="MuiFormControl-root MuiTextField-root"
>
  <label
    class="MuiFormLabel-root MuiInputLabel-root makeStyles-root-3 MuiInputLabel-formControl MuiInputLabel-animated"
    data-shrink="false"
    for="jest"
    id="jest-label"
  >
    jest
  </label>
  <div
    class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-underline-2 MuiInputBase-formControl MuiInput-formControl"
  >
    <input
      aria-invalid="false"
      class="MuiInputBase-input MuiInput-input makeStyles-input-1"
      id="jest"
      type="text"
      value=""
    />
  </div>
</div>
`;
